src/js/other-websites/inpage_menu.js

```diff
@@ -51,226 +52,615 @@ async function isUserSignedIn() {
-    // but the user has already reached the max number of aliases.
-    if (newRelayAddressResponse.status === 402) {
-      relayInPageMenu.classList.remove("fx-relay-alias-loading");
-      // preserve menu height before removing child elements
-      // TODO: Add background/function to adjust height of iframe
-      relayInPageMenu.style.height = relayInPageMenu.clientHeight + "px";
+    // Focus on "Go to Firefox Relay" button
+    signUpButton.focus();
+
+    document.getElementById("fxRelayMenuBody").classList.remove("is-loading");
 
-      [generateAliasBtn, remainingAliasesSpan].forEach((el) => {
-        el.remove();
+    // Bug: There's a race condition on how fast to detect the iframe being loaded. The setTimeout solves it for now.
+    setTimeout(async () => {
+      await browser.runtime.sendMessage({
+        method: "updateIframeHeight",
+        height: document.getElementById("fxRelayMenuBody").scrollHeight,
       });
+    }, 10);
+  },
+  components: {
+    generateMaskButton: async () => {
+      // Create "Generate Relay Address" button
+      const generateAliasBtn = document.querySelector(
+        ".fx-relay-menu-generate-alias-btn"
+      );
 
-      const errorMessage = document.createElement("p");
-      errorMessage.classList.add("fx-relay-error-message");
+      generateAliasBtn.textContent = browser.i18n.getMessage(
+        "pageInputIconGenerateNewAlias_mask"
+      );
 
-      errorMessage.textContent = browser.i18n.getMessage(
-        "pageInputIconMaxAliasesError_mask",
-        [relayAddresses.length]
+      // Handle "Generate New Alias" clicks
+      generateAliasBtn.addEventListener("click", async (generateClickEvt) => {
+        sendInPageEvent("click", "input-menu-reuse-previous-alias");
+        preventDefaultBehavior(generateClickEvt);
+
+        generateAliasBtn.classList.add("is-loading");
+
+        // Request the active tab from the background script and parse the `document.location.hostname`
+        const currentPageHostName = await browser.runtime.sendMessage({
+          method: "getCurrentPageHostname",
+        });
+
+        // Attempt to create a new alias
+        const newRelayAddressResponse = await browser.runtime.sendMessage({
+          method: "makeRelayAddress",
+          description: currentPageHostName,
+        });
+
+        // TODO: Determine if this check is still necessary. Note that there's no error state defined currently.
+
+        // Catch edge cases where the "Generate New Alias" button is still enabled,
+        // but the user has already reached the max number of aliases.
+        if (newRelayAddressResponse.status === 402) {
+          // relayInPageMenu.classList.remove("fx-relay-alias-loading");
+          // // preserve menu height before removing child elements
+          // relayInPageMenu.style.height = relayInPageMenu.clientHeight + "px";
+
+          // [generateAliasBtn, remainingAliasesSpan].forEach((el) => {
+          //   el.remove();
+          // });
```

Comment:
question (non-blocking): Does this block of commented code need to stay for a follow-up?

Reply:
Removed in 72a0b6897317b4c0f420202b273dff4f73da538a